src/c++/vdo/base/message-stats.c

```diff
@@ -430,3 +432,59 @@ int vdo_write_stats(struct vdo *vdo, char *buf, unsigned int maxlen)
 	vdo_free(stats);
 	return VDO_SUCCESS;
 }
+
+static void write_index_memory(char *prefix, u32 mem, char *suffix,
+			       char **buf, unsigned int *maxlen)
+{
+	/* Determine if the index memory value is fractional */
+	if (mem > 1024) {
+		/* Convert index memory to fractional value */
```

Comment:
I think this will read cleaner (and be simpler) if you check for the three special values first, and otherwise print out the bare number. (The 1024 value is also kind of magical here, but you shouldn't need it if you check the special values first).

Reply:
This was how vdo manager did it, so I was just following the pattern. I can rework it if you think its a big deal